File: A_Kevin_and_Combination_Lock.cpp

#include <iostream>
#include <vector>
#include <string>
using namespace std;

void solve() {
    int n;
    cin >> n;
    while (n--) {
        int number;
        cin >> number;
        while (number > 0) {
            string str = to_string(number);
            vector<int> digits;
            for (char c : str) {
                digits.push_back(c - '0');
            }

            bool modified = false;
            for (int j = 0; j < digits.size() - 1; j++) {
                if (digits[j] == 3 && digits[j + 1] == 3) {
                    digits.erase(digits.begin() + j, digits.begin() + j + 2);
                    modified = true;
                    break;
                }
            }

            if (modified) {
                number = 0;
                for (int digit : digits) {
                    number = number * 10 + digit;
                }
            } else if (number >= 33) {
                number -= 33;
            } else {
                break;
            }
        }

        if (number == 0) {
            cout << "YES" << endl;
        } else {
            cout << "NO" << endl;
        }
    }
}

int main() 
{
    solve();
    return 0;

}

File: A_Kevin_and_Combination_Lock.py
def solve():
    n = int(input())  
    for _ in range(n):
        number = int(input())  
        while number > 0:
            digits = [int(digit) for digit in str(number)]
            modified = False  
            
            for j in range(len(digits) - 1):
                if digits[j] == digits[j + 1] == 3:
                    digits = digits[:j] + digits[j + 2:]
                    modified = True
                    break
            
            if modified:
                number = int("".join(map(str, digits))) if digits else 0
            elif number >= 33:
                number -= 33
            else:
                break

        if number == 0:
            print("YES")
        else:
            print("NO")

solve()
